lib/src/Base/Algo/CubaIntegration.cxx

```diff
@@ -116,8 +116,21 @@ Point CubaIntegration::integrate(const Function & function,
   const int flags = ResourceMap::GetAsUnsignedInteger("CubaIntegration-flags");
   const int seed = ResourceMap::GetAsUnsignedInteger("CubaIntegration-seed"); /* The seed for the pseudo-random-number generator */
 
+  if (inputDimension > 1024)
+      throw InvalidArgumentException(HERE) << "Cuba maximum input dimension is 1024, got " << inputDimension;
+  if (outputDimension > 1024)
+      throw InvalidArgumentException(HERE) << "Cuba maximum output dimension is 1024, got " << outputDimension;
+
+  // fallback for dimension < 2
+  String algorithmName = algorithmName_;
+  if ((inputDimension == 1) && Description({"cuhre", "divonne"}).contains(algorithmName))
+  {
+    algorithmName = "vegas";
+    LOGWARN(OSS() << "Cuba algorithm switched from "<< algorithmName_ << " to " << algorithmName);
```

Comment:
Shouldn't it throw rather than override the value stored in an attribute?

Reply:
yes its bad